contractcourt/chain_watcher.go

```diff
@@ -377,23 +377,44 @@ func (c *chainWatcher) closeObserver(spendNtfn *chainntnfs.SpendEvent) {
 			)
 			remoteStateNum := remoteCommit.CommitHeight
 
+			remoteChainTip, err := c.chanState.RemoteCommitChainTip()
+			if err != nil && err != channeldb.ErrNoPendingCommit {
+				log.Errorf("unable to obtain chain tip for "+
+					"ChannelPoint(%v): %v",
+					c.chanState.FundingOutpoint, err)
+				return
+			}
+
 			switch {
 			// If state number spending transaction matches the
 			// current latest state, then they've initiated a
 			// unilateral close. So we'll trigger the unilateral
 			// close signal so subscribers can clean up the state
 			// as necessary.
-			//
+			case broadcastStateNum == remoteStateNum:
+				err := c.dispatchRemoteForceClose(
+					commitSpend, *remoteCommit, false,
+				)
+				if err != nil {
+					log.Errorf("unable to handle remote "+
+						"close for chan_point=%v: %v",
+						c.chanState.FundingOutpoint, err)
+				}
+
 			// We'll also handle the case of the remote party
 			// broadcasting their commitment transaction which is
 			// one height above ours. This case can arise when we
 			// initiate a state transition, but the remote party
 			// has a fail crash _after_ accepting the new state,
 			// but _before_ sending their signature to us.
-			case broadcastStateNum >= remoteStateNum:
-				if err := c.dispatchRemoteForceClose(
-					commitSpend, *remoteCommit,
-				); err != nil {
+			case broadcastStateNum == remoteStateNum+1 &&
```

Comment:
maybe add warning in the even that `broadcastStateNum > remoteStateNum+1` now that we would take no action?

Reply:
Fixed.